sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayImpl.java

```diff
@@ -428,6 +444,25 @@ protected ApplicationGatewayBackendImpl ensureUniqueBackend() {
         return backend;
     }
 
+    private void ensureNoMixedWaf() {
+        String errorMessage = "Can't have a mixture of legacy WAF configuration and WAF policy. "
```

Comment:
nit, maybe "A mixture of legacy WAF configuration and WAF policy is not allowed."